src/axolotl/utils/data.py

```diff
@@ -905,3 +906,30 @@ def encode_packed_pretraining(
             chunked_data[feature].append(collated_features[feature].squeeze(0))
 
     return chunked_data
+
+
+def load_prepare_dpo_datasets(cfg):
+    def load_split(dataset_cfgs, _cfg):
+        split_datasets: List[Any] = []
+        for i, ds_cfg in enumerate(dataset_cfgs):
+            ds = load_dataset(  # pylint: disable=invalid-name
+                ds_cfg["path"], split=ds_cfg["split"]
+            )
+            split_datasets.insert(i, ds)
+
+        for i, data_set in enumerate(split_datasets):
+            _type = dataset_cfgs[i]["type"]
+            ds_transform_fn = load_dpo(_type, _cfg)
+            split_datasets[i] = data_set.map(ds_transform_fn)
```

Comment:
```suggestion
            if _type:
                ds_transform_fn = load_dpo(_type, _cfg)
                split_datasets[i] = data_set.map(ds_transform_fn)
            else:
                # If no `type` is provided, assume the dataset is already in the expected format with 
                # "prompt", "chosen" and "rejected" already preprocessed
                split_datasets[i] = data_set
```

Reply:
In case the `type` is not informed in the config file, it could be assumed the dataset is already preprocessed and it shouldn't be needed to apply the `ds_transform_fn`. This way the user has freedom to pass any type of dataset, assuming it's already preprocessed (maybe informing via `LOG.info` would help keeping track).